pkg/monitor/cluster/nodes.go

```diff
@@ -0,0 +1,57 @@
+package cluster
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type metricsLabel string
+
+const (
+	notReadyNodes           metricsLabel = "notReadyNodes"
+	memoryPressureNodes     metricsLabel = "memoryPressureNodes"
+	diskPressureNodes       metricsLabel = "diskPressureNodes"
+	pidPressureNodes        metricsLabel = "pidPressureNodes"
+	networkUnavailableNodes metricsLabel = "networkUnavailableNodes"
+)
+
+var metricsLabelMap = map[corev1.NodeConditionType]metricsLabel{
+	corev1.NodeReady:              notReadyNodes,
+	corev1.NodeMemoryPressure:     memoryPressureNodes,
+	corev1.NodeDiskPressure:       diskPressureNodes,
+	corev1.NodePIDPressure:        pidPressureNodes,
+	corev1.NodeNetworkUnavailable: networkUnavailableNodes,
+}
+
+func (mon *Monitor) emitNodesMetrics() error {
+	nodes, err := mon.cli.CoreV1().Nodes().List(metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+	mon.emitGauge("cluster.nodes.total", int64(len(nodes.Items)), map[string]string{})
```

Comment:
```suggestion
	mon.emitGauge("cluster.nodes.total", int64(len(nodes.Items)), nil)
```

Reply:
this can't be nil. Unless we change this chain of functions that pass dims and iterate over it https://github.com/Azure/ARO-RP/blob/8b57b7cda9b089547a6b921ef199074b7337026c/pkg/monitor/cluster/cluster.go#L87

Reply:
good catch - please update `func (mon *Monitor) emit{Float,Gauge}` so they allocate in the case of a nil dims input.